R/submit-model.R

```diff
@@ -36,12 +44,7 @@ submit_model.bbi_nmbayes_model <- function(
     fs::path_abs(.config_path)
   }
 
-  # Convert model to bbi_nonmem_model for initialization. Another option would
-  # be to call NextMethod(), but modifying arguments with that approach less
-  # straightforward.
-  mod_init <- .mod
-  class(mod_init) <- class(mod_init)[-1]
-
+  mod_init <- nmbayes_init(.mod, .overwrite)
```

Comment:
I'm not exactly sure how to handle this, and I'm willing to concede that maybe we should punt on it and address it in #35 , but this is strange behavior to me: 

## Unsuccessful model submission

an unsuccessful model submission (in my case because my data path was wrong) still results in the inits getting created.

```r
# unsuccessful submission
submit_model(new1002, .dry_run = TRUE)
## Error in check_status_code(p$get_exit_status(), output, .cmd_args) :                                                                                                                
## ...
## msg="An error occurred during model processing: unable to open datafile at /data/home/sethg/tools-testing/data/derived/analysis3.csv ..."

# me fixing it...
data_dir <- fs::path_abs(file.path(MD, "../../data/derived"))
fs::file_copy("~/example-projects/bbr-nonmem-poppk-bayes/data/derived/analysis3.csv", data_dir)

# me trying to resubmit
submit_model(new1002)
## Error in nmbayes_init(.mod, .overwrite) : 
##   init submodel already exists for 1002

fs::dir_ls(get_output_dir(new1002))
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/init.ctl
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/init.yaml
```

## Dry run submission

I'm pretty sure this is the known behavior documented in #35 , but it does seem unfortunate that running a dry run then causes an error when you try to submit for real:

```r
submit_model(new1002, .dry_run = TRUE)
fs::dir_ls(get_output_dir(new1002))
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/1002-1.ctl
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/1002-1.yaml
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/1002-2.ctl
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/1002-2.yaml
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/1002-3.ctl
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/1002-3.yaml
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/1002-4.ctl
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/1002-4.yaml
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/init
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/init.chn
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/init.ctl
## /data/home/sethg/tools-testing/nmbayes-initial-testing/model/1002/init.yaml


submit_model(new1002)
## Error in nmbayes_init(.mod, .overwrite) : 
##   init submodel already exists for 1002
```

Should we punt on both of these and fix later, or is there an in-between state that would be a) relatively easy to implement and b) less weird for the user than this?

Reply:
> Should we punt on both of these and fix later, or is there an in-between state that would be a) relatively easy to implement and b) less weird for the user than this?

Knowing that gh-35 was something we need to get to, I was being lazy here and not thinking about the handling really at all.  But you're probably right that there's a better intermediate state to leave it at then where I did.  I'll think about it a bit.

Reply:
944a214 should address these two cases (i.e. a previous dry run or "no output" failed run won't require `.overwrite = TRUE`).

Reply:
Hmm, nothing jumps out to me.